logical/response.go

```diff
@@ -110,3 +110,18 @@ func ListResponse(keys []string) *Response {
 	}
 	return resp
 }
+
+// ListResponseWithInfo is used to format a response to a list operation and
+// return the keys as well as a map with corresponding key info.
+func ListResponseWithInfo(keys []string, keyInfo map[string]interface{}) *Response {
+	resp := ListResponse(keys)
+
+	resp.Data["key_info"] = map[string]interface{}{}
```

Comment:
You should create the map separately and assign to resp.Data at the end of the response. This avoids having to keep type asserting in the loop...cleaner _and_ faster!

Separately, I'm not sure whether, if there are no actual keys, we should return an empty map, a null value, or not add the key at all.

Reply:
I went with not returning "key_info" at all if it's empty, similar to the behavior with "keys".

Reply:
Sounds good.